openmdao/recorders/base_recorder.py

```diff
@@ -258,6 +258,45 @@ def record_iteration_problem(self, recording_requester, data, metadata):
         """
         raise NotImplementedError("record_iteration_problem has not been overridden")
 
+    def record_derivatives(self, recording_requester, data, metadata, **kwargs):
+        """
+        Route the record_derivatives call to the proper method.
+
+        Parameters
+        ----------
+        recording_requester : object
+            System, Solver, Driver in need of recording.
+        data : dict
+            Dictionary containing derivatives keyed by 'of,wrt' to be recorded.
+        metadata : dict
+            Dictionary containing execution metadata.
+        **kwargs : keyword args
+            Some implementations of record_derivatives need additional args.
+        """
+        if not self._parallel:
+            if MPI and MPI.COMM_WORLD.rank > 0:
+                raise RuntimeError("Non-parallel recorders should not be recording on ranks > 0")
+
+        self._iteration_coordinate = recording_iteration.get_formatted_iteration_coordinate()
+
+        if isinstance(recording_requester, Driver):
+            self.record_derivatives_driver(recording_requester, data, metadata)
```

Comment:
seems like there should be an `else raise RuntimeError` here...

Reply:
I only call this function from a Driver. Probably don't even need the "if", but I was keeping the code in parallel with the other recorders.  I will just take that out to eliminate confusion.

Reply:
but the user could incorrectly call it from another object, in which case we want it to error, right?

Reply:
Can you give me a usecase (code example) where a user calls record_derivatives?

Reply:
Or record_iteration if you are more familiar with that method.  We currently have not coverage for the "else raise RuntimeError" branch on that.

Reply:
sorry you're right.. ignore last comment